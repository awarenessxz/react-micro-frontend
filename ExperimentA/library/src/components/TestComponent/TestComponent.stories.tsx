import * as React from 'react';
import { Meta } from '@storybook/react/types-6-0';
import TestComponent from './TestComponent';

export default {
    title: 'Components/TestComponent',
    component: TestComponent,
} as Meta;

export const Primary = (): JSX.Element => <TestComponent theme="primary" />;

export const Secondary = (): JSX.Element => <TestComponent theme="secondary" />;
