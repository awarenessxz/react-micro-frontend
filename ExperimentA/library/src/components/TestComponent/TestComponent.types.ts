export interface TestComponentProps {
    /**
     * Theme of Test component
     */
    theme: 'primary' | 'secondary';
}
