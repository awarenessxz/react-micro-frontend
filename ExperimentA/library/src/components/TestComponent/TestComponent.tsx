import React from 'react';
import { TestComponentProps } from './TestComponent.types';
import styles from './TestComponent.module.scss';

/**
 * Test Component
 */
const TestComponent = ({ theme }: TestComponentProps): JSX.Element => {
    const mThemeStyle = theme === 'primary' ? '' : styles.testComponentSecondary;

    return (
        <div data-testid="test-component" className={`${styles.testComponent} ${mThemeStyle}`}>
            <h1 className={`${styles.heading}`}>I&apos;m the test component</h1>
        </div>
    );
};

export default TestComponent;
