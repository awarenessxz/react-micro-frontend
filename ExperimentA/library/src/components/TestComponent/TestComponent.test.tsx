import * as React from 'react';
import { render } from '@testing-library/react';

// Import Component
import TestComponent from './TestComponent';
import { TestComponentProps } from './TestComponent.types';

// function to render Component before each test
const renderComponent = (props = {}) => {
    const defaultProps: TestComponentProps = {
        theme: 'primary',
    };
    const merged = { ...defaultProps, ...props };
    return render(<TestComponent {...merged} />);
};

// 1. Testing if component renders properly
describe('Testing if component renders properly', () => {
    // Snapshot Testing
    it('Snapshot Testing', () => {
        const { asFragment } = renderComponent();
        expect(asFragment()).toMatchSnapshot();
    });

    // Testing Dom Elements: check if renders with correct value (Note: the usage of data-testid in component)
    describe('Testing if component renders with correct state/value', () => {
        it('should have primary className with default props', () => {
            const { getByTestId } = renderComponent();
            const testComponent = getByTestId('test-component');
            expect(testComponent).not.toHaveClass('testComponentSecondary');
        });

        it('should have secondary className with theme set as secondary', () => {
            const { getByTestId } = renderComponent({ theme: 'secondary' });
            const testComponent = getByTestId('test-component');
            expect(testComponent).toHaveClass('testComponentSecondary');
        });
    });
});
