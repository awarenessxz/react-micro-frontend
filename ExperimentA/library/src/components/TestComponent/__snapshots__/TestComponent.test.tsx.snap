// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing if component renders properly Snapshot Testing 1`] = `
<DocumentFragment>
  <div
    class="testComponent "
    data-testid="test-component"
  >
    <h1
      class="heading"
    >
      I'm the test component
    </h1>
    <h2>
      Made with love by Harvey
    </h2>
  </div>
</DocumentFragment>
`;
